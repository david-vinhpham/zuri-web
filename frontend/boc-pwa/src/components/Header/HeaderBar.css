/**
 * Home Credit (https://www.homecredit.vn/)
 *
 * Copyright Â© 2018-present HomeCredit, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

@import '../variables.css';

:root {
  --title-font-weight: 400;
}

.header {
  background-color: var(--c-accent-color);
  display: flex;
  align-items: center;

  & > span {
    flex: 1;
  }

  & > span:nth-of-type(1) {
    text-align: left;
  }

  & > span:nth-of-type(2) {
    text-align: center;
  }

  & > span:nth-of-type(3) {
    text-align: right;
  }
}

.title {
  width: 100%;
  margin: 0;
  text-align: center;
  background-color: var(--c-accent-color);
  font-weight: var(--title-font-weight);
  font-size: var(--f-label);
  padding: var(--b-space-md) 0;
  color: var(--c-ui-lighter);
  font-style: italic;
}

.disableText {
  color: var(--c-text-lighter);
}

.nextText {
  padding-right: var(--b-space-2xs);
}

.btnNext {
  box-shadow: none;
  padding-right: var(--b-space-xs);
}

.btnNext:active {
  background-color: transparent;
}
