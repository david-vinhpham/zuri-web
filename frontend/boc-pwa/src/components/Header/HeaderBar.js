import React, { Component } from 'react';
import { string, bool, func } from 'prop-types';
import classNames from 'classnames';
import { IconButton, Button } from 'homecredit-ui';
import noop from 'lodash/noop';

import withStyles from 'isomorphic-style-loader/lib/withStyles';
import s from './HeaderBar.css';

const BOCVNIcon = () => (
  <svg fill="#fff" viewBox="0 -40 153 153">
    <g>
      <path d="M76.503 0C34.255 0 0 32.96 0 73.637c0 14.343 4.261 27.734 11.634 39.052H48.53c-11.201-8.566-18.477-22.111-18.477-37.405 0-25.918 20.793-46.923 46.443-46.923s46.444 21.005 46.444 46.923c0 15.294-7.266 28.84-18.477 37.405h36.902C148.744 101.37 153 87.98 153 73.637 153 32.96 118.76 0 76.503 0" />
      <path d="M50.39 87.036c3.548 11.73 13.276 20.156 24.73 20.156 11.465 0 21.203-8.425 24.742-20.156H50.39" />
    </g>
  </svg>
);

class HeaderBar extends Component {
  static propTypes = {
    title: string,
    hideBackButton: bool,
    hideNextButton: bool,
    onBackClick: func,
    onNextClick: func,
    nextText: string,
    disabledBackButton: bool,
    disabledNextButton: bool,
    className: string,
  };

  static defaultProps = {
    title: 'Hello http://www.bocvietnam.com',
    hideBackButton: false,
    hideNextButton: false,
    onBackClick: noop,
    onNextClick: noop,
    nextText: 'Tiếp tục',
    disabledBackButton: false,
    disabledNextButton: false,
    className: '',
  };

  render() {
    const {
      title,
      hideBackButton,
      hideNextButton,
      nextText,
      onBackClick,
      onNextClick,
      disabledBackButton,
      disabledNextButton,
      className,
    } = this.props;
    const nextTextClass = classNames(s.nextText, {
      [s.disableText]: disabledNextButton,
    });

    return (
      <div className={className}>
        <header className={s.header}>
          <span>
            {!hideBackButton && (
              <IconButton
                smallIcon
                icon="arrow_back"
                onClick={onBackClick}
                disabled={disabledBackButton}
              />
            )}
          </span>
          <span>
            <IconButton icon={<BOCVNIcon />} />
          </span>
          <span>
            {!hideNextButton && (
              <Button
                onClick={onNextClick}
                disabled={disabledNextButton}
                className={s.btnNext}
              >
                <span className={nextTextClass}>{nextText}</span>
              </Button>
            )}
          </span>
        </header>
        {title && <h1 className={s.title}>{title}</h1>}
      </div>
    );
  }
}

export default withStyles(s)(HeaderBar);
