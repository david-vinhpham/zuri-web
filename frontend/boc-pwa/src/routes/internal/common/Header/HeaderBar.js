import React, { Component } from 'react';
import { string, bool, func } from 'prop-types';
import classNames from 'classnames';
import { IconButton, Button } from 'homecredit-ui';
import noop from 'lodash/noop';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import Logo from '../Logo';
import UserHeader from './UserHeader';
import s from './HeaderBar.css';

class HeaderBar extends Component {
  static propTypes = {
    title: string,
    hideBackButton: bool,
    hideNextButton: bool,
    onBackClick: func,
    onNextClick: func,
    nextText: string,
    disabledBackButton: bool,
    disabledNextButton: bool,
    className: string,
    isCheckAuth: bool,
  };

  static defaultProps = {
    title: 'Hello http://www.bocvietnam.com',
    hideBackButton: false,
    hideNextButton: false,
    onBackClick: noop,
    onNextClick: noop,
    nextText: 'Tiếp tục',
    disabledBackButton: false,
    disabledNextButton: false,
    className: '',
    isCheckAuth: false,
  };

  render() {
    const {
      title,
      hideBackButton,
      hideNextButton,
      nextText,
      onBackClick,
      onNextClick,
      disabledBackButton,
      disabledNextButton,
      className,
      isCheckAuth,
    } = this.props;
    const nextTextClass = classNames(s.nextText, {
      [s.disableText]: disabledNextButton,
    });

    return (
      <div className={className}>
        <header className={s.header}>
          <span>
            {!hideBackButton && (
              <IconButton
                smallIcon
                icon="arrow_back"
                onClick={onBackClick}
                disabled={disabledBackButton}
              />
            )}
          </span>
          <span>
            <IconButton icon={<Logo />} />
          </span>
          <span>
            {!hideNextButton && (
              <Button
                onClick={onNextClick}
                disabled={disabledNextButton}
                className={s.btnNext}
              >
                <span className={nextTextClass}>{nextText}</span>
              </Button>
            )}
            <UserHeader isCheckAuth={isCheckAuth} />
          </span>
        </header>
        {title && <h1 className={s.title}>{title}</h1>}
      </div>
    );
  }
}

export default withStyles(s)(HeaderBar);
