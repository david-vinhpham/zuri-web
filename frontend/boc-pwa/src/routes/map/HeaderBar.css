@import '../../components/variables.css';

:root {
  --icon-width: 34px;
  --header-height: 50px;
}

.headerBar {
  background-color: var(--c-accent-color);
  text-align: center;
  height: var(--header-height);
  display: flex;
  justify-content: center;
  align-items: center;
}

.iconLogo {
  width: var(--icon-width);
}
