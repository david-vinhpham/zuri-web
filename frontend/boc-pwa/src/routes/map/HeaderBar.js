import React, { Component } from 'react';
import withStyles from 'isomorphic-style-loader/lib/withStyles';

import Link from '../../components/Link';
import s from './HeaderBar.css';

const HomecreditIcon = () => (
  <svg className={s.iconLogo} fill="#fff" viewBox="0 -14 153 153">
    <g>
      <path d="M76.503 0C34.255 0 0 32.96 0 73.637c0 14.343 4.261 27.734 11.634 39.052H48.53c-11.201-8.566-18.477-22.111-18.477-37.405 0-25.918 20.793-46.923 46.443-46.923s46.444 21.005 46.444 46.923c0 15.294-7.266 28.84-18.477 37.405h36.902C148.744 101.37 153 87.98 153 73.637 153 32.96 118.76 0 76.503 0" />
      <path d="M50.39 87.036c3.548 11.73 13.276 20.156 24.73 20.156 11.465 0 21.203-8.425 24.742-20.156H50.39" />
    </g>
  </svg>
);

class HeaderBar extends Component {
  render() {
    return (
      <div className={s.headerBar}>
        <Link to="/">
          <HomecreditIcon />
        </Link>
      </div>
    );
  }
}

export default withStyles(s)(HeaderBar);
