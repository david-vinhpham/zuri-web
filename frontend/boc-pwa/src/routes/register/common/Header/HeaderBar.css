/**
 * BOC VN (http://www.bocvietnam.com/)
 *
 * Copyright Â© 2019-present BOCVN, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

@import '../../../../components/variables.css';

:root {
  --title-font-weight: 400;
}

.header {
  background-color: var(--c-accent-color);
  display: flex;
  align-items: center;

  & > span {
    flex: 1;
  }

  & > span:nth-of-type(1) {
    text-align: left;
  }

  & > span:nth-of-type(2) {
    text-align: center;
  }

  & > span:nth-of-type(3) {
    text-align: right;
  }
}

.title {
  width: 100%;
  margin: 0;
  text-align: center;
  background-color: var(--c-accent-color);
  font-weight: var(--title-font-weight);
  font-size: var(--f-label);
  padding: var(--b-space-md) 0;
  color: var(--c-ui-lighter);
  font-style: italic;
}
