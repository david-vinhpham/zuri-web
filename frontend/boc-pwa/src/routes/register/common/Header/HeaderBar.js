import React, { Component } from 'react';
import { string } from 'prop-types';
import { IconButton } from 'homecredit-ui';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import Logo from '../Logo';
import UserHeader from './UserHeader';
import Auth from '../../common/Auth';
import s from './HeaderBar.css';

class HeaderBar extends Component {
  static propTypes = {
    title: string,
    className: string,
  };

  static defaultProps = {
    title: '',
    className: '',
  };

  render() {
    const { title, className } = this.props;

    return (
      <div className={className}>
        <header className={s.header}>
          <span />
          <span>
            <IconButton icon={<Logo />} />
          </span>
          <span>
            <Auth
              renderAuth={({ username }, logout) => (
                <UserHeader username={username} logout={logout} />
              )}
            />
          </span>
        </header>
        {title && <h1 className={s.title}>{title}</h1>}
      </div>
    );
  }
}

export default withStyles(s)(HeaderBar);
