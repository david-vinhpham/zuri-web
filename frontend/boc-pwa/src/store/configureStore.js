import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { persistStore, persistReducer } from 'redux-persist';

import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';

import storage from 'redux-persist/lib/storage';
import { name, version } from '../../package.json';
import rootReducer from '../reducers';
import createLogger from './logger';

export default function configureStore(initialState, helpersConfig) {
  const middleware = [thunk.withExtraArgument(helpersConfig)];

  const persistConfig = {
    key: 'BOCVN',
    storage,
    stateReconciler: autoMergeLevel2,
    blacklist: ['clz'],
  };

  const persistedReducer = persistReducer(persistConfig, rootReducer);

  let enhancer;

  if (__DEV__) {
    middleware.push(createLogger());

    // https://github.com/zalmoxisus/redux-devtools-extension#14-using-in-production
    const composeEnhancers = composeWithDevTools({
      // Options: https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#options
      name: `${name}@${version}`,
    });

    // https://redux.js.org/docs/api/applyMiddleware.html
    enhancer = composeEnhancers(applyMiddleware(...middleware));
  } else {
    enhancer = applyMiddleware(...middleware);
  }

  // https://redux.js.org/docs/api/createStore.html
  const store = createStore(persistedReducer, enhancer);

  // Hot reload reducers (requires Webpack or Browserify HMR to be enabled)
  if (__DEV__ && module.hot) {
    module.hot.accept('../reducers', () =>
      // Don't forget to remove `()` if you change reducers back to normal rootReducer.
      // eslint-disable-next-line global-require
      store.replaceReducer(require('../reducers').default()),
    );
  }

  const persistor = persistStore(store);

  return { store, persistor };
}
