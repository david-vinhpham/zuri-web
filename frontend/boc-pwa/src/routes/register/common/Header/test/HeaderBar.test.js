import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import HeaderBar from '../HeaderBar';

const mockStore = configureStore();
const initialState = {
  saRegister: {
    auth: {
      username: null,
    },
  },
};

it('HeaderBar snapshot', () => {
  const store = mockStore(initialState);
  const wrapper = renderer.create(
    <Provider store={store}>
      <HeaderBar />
    </Provider>,
  );
  const snapshot = wrapper.toJSON();

  expect(snapshot).toMatchSnapshot();
});
