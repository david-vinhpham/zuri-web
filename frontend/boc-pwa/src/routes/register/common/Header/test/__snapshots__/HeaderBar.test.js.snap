// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar snapshot 1`] = `
<div
  className=""
>
  <header
    className="header"
  >
    <span />
    <span>
      <button
        className="IconButton_toggle__QueJK IconButton_button__2pkKY IconButton_neutral__3FoxH"
        disabled={false}
        href={null}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
      >
        <span
          className="IconButton_icon__1sSHF"
        >
          <svg
            fill="#fff"
            viewBox="0 -40 153 153"
          >
            <g>
              <path
                d="M76.503 0C34.255 0 0 32.96 0 73.637c0 14.343 4.261 27.734 11.634 39.052H48.53c-11.201-8.566-18.477-22.111-18.477-37.405 0-25.918 20.793-46.923 46.443-46.923s46.444 21.005 46.444 46.923c0 15.294-7.266 28.84-18.477 37.405h36.902C148.744 101.37 153 87.98 153 73.637 153 32.96 118.76 0 76.503 0"
              />
              <path
                d="M50.39 87.036c3.548 11.73 13.276 20.156 24.73 20.156 11.465 0 21.203-8.425 24.742-20.156H50.39"
              />
            </g>
          </svg>
        </span>
      </button>
    </span>
    <span />
  </header>
  
</div>
`;
